### PR TITLE
Use `debian:bullseye` as the Dockerfile base image

### PR DESCRIPTION
**Pull request by omus**
Continuing the work from #197. By moving away from using the `rayproject/ray` image as a base to the more generic `debian` image we should be able to more easily support basing images off of other images such as `cuda` for GPU support.

As a bonus side effect this reduces the image size from [4.52GB](https://github.com/beacon-biosignals/Ray.jl/actions/runs/6499378331?pr=197) to [1.38GB](https://github.com/beacon-biosignals/Ray.jl/actions/runs/6500584043?pr=199).

### CODE REVIEW

**Comment by omus** [comment]
Seeing this failure on the RayCluster when using this image:
```
$ kubectl logs pod/raycluster-head-2pr8c -c autoscaler
Traceback (most recent call last):
  File "/home/ray/.local/bin/ray", line 8, in <module>
    sys.exit(main())
  File "/home/ray/.local/lib/python3.10/site-packages/ray/scripts/scripts.py", line 2462, in main
    return cli()
  File "/home/ray/.local/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/ray/.local/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/ray/.local/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ray/.local/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ray/.local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/ray/.local/lib/python3.10/site-packages/ray/scripts/scripts.py", line 2194, in kuberay_autoscaler
    from ray.autoscaler._private.kuberay.run_autoscaler import run_kuberay_autoscaler
  File "/home/ray/.local/lib/python3.10/site-packages/ray/autoscaler/_private/kuberay/run_autoscaler.py", line 11, in <module>
    from ray.autoscaler._private.kuberay.autoscaling_config import AutoscalingConfigProducer
  File "/home/ray/.local/lib/python3.10/site-packages/ray/autoscaler/_private/kuberay/autoscaling_config.py", line 7, in <module>
    import kubernetes
ModuleNotFoundError: No module named 'kubernetes'
```

**Comment by omus** [comment]
Dashboard is broken. Reports messages like:
```
2023-10-12 19:17:10,661	INFO utils.py:123 -- Module ray.dashboard.modules.actor.actor_head cannot be loaded because we cannot import all dependencies. Install this module using `pip install 'ray[default]'` for the full dashboard functionality. Error: No module named 'aiohttp'
```

**Comment by omus** [comment]
When submitting a job I was seeing `/bin/sh: 1: julia: not found`. Showing the `PATH` for the submitted job revealed the path was "/home/ray/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games" instead of what I was seeing interactively: "/home/ray/.local/bin:/usr/local/julia/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin". I'm not sure why this difference exists but I'll solve this by symlinking "/usr/local/julia/bin/julia" to "/usr/local/bin/julia" instead of adding the Julia bin directory to the `PATH`.

**Comment by omus** [comment]
I've validated that this Dockerfile works with the private RayCluster experiment we have.

File: Dockerfile [hunk 1]
```diff
@@ -37,6 +52,21 @@ RUN if ! julia --history-file=no -e 'exit(0)'; then \
 # Reduces output from `apt-get`
 ENV DEBIAN_FRONTEND="noninteractive"
 
+# Install sudo
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get -qq update && \
+    apt-get -qq install sudo && \
+    echo "%sudo ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
```
Comment (glennmoy) [inline Dockerfile:60]: what's this do?

File: Dockerfile [hunk 2]
```diff
@@ -236,12 +277,16 @@ RUN julia --project=${RAY_JL_PROJECT} -e 'using Pkg; Pkg.precompile(strict=true,
...
+COPY --from=build-ray-jl --link ${PYTHON_USER_BASE} ${PYTHON_USER_BASE}
+ENV PATH="${PYTHON_USER_BASE}/bin:${PATH}"
+RUN ray --version
 
 ARG RAY_JL_PROJECT=${JULIA_USER_DEPOT}/dev/Ray
+COPY --from=build-ray-jl --chown=${UID} --link ${JULIA_USER_DEPOT} ${JULIA_USER_DEPOT}
+RUN julia --project=${RAY_JL_PROJECT} -e 'using Ray'
 
 # Set up default project and working dir so that users need only pass in the requisite script input args
 ENV JULIA_PROJECT=${RAY_JL_PROJECT}
```
Comment (glennmoy) [inline Dockerfile:292]: why not set the ENV and WORKDIR before calling `RUN julia --project=${RAY_JL_PROECT}`?

**Review by glennmoy — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (omus) [reply Dockerfile:65]: Allows users without passwords to use `sudo`

File: Dockerfile (hunk 2 above)
Comment (omus) [reply Dockerfile:297]: We could. As this Dockerfile uses multiple projects I think it's clearer to not use `JULIA_PROJECT` within this Dockerfile. Setting these at the end are just conveniences for interactive Docker image use.

**Comment by omus** [comment]
Doing one last validation on this before merging

**Comment by omus** [comment]
Validated that the latest changes still work

**Comment by omus** [comment]
Validated this is working with the RayCluster setup

**Pull request merged**